addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -365,10 +365,9 @@ export class WebsitePreview extends Component {
                 const destinationUrl = new URL(href, window.location);
                 destinationUrl.searchParams.delete('edit_translations');
                 destinationUrl.hash = this.websiteService.contentWindow.location.hash;
-                const forceLangUrl = `/website/lang/${lang}?r=${destinationUrl.toString()}`;
                 this.websiteService.bus.trigger('LEAVE-EDIT-MODE', {
                     onLeave: () => {
-                        this.websiteService.goToWebsite({ path: forceLangUrl });
+                        this.websiteService.goToWebsite({ path: destinationUrl.toString(), lang });
```

Comment:
:nauseated_face: I hate this _object property shorthands_ notation so much, but that's how it is I guess, even tho I think mixing it with the regular notation in a same object construction is atrocious
Just my 2 cents, ignore this :innocent: